model_read_only/migrations/14.0.2.0.0/post-migration.py

```diff
@@ -0,0 +1,29 @@
+from collections import defaultdict
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            SELECT ir_model_id, res_groups_id
+            FROM ir_model_group_readonly_rel_tmp;
+        """,
+    )
+    data = env.cr.fetchall()
+    grouped = defaultdict(list)
+    for key, value in data:
+        grouped[key].append(value)
+    vals_list = [
+        {"model_id": model_id, "group_ids": group_ids}
+        for model_id, group_ids in grouped
```

Comment:
Grouped is a dict variable that should use `.item()` for iterating dict items. I got around this point in the last comment, sorry for that. Than your code must be looks like this:
```py
        for model_id, group_ids in grouped.items()
```